plugins/org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/ui/trace/ZipFileAwareTrace.java

```diff
@@ -68,4 +72,12 @@ public InputStream getContents(SourceRelativeURI uri, IProject project) {
 		return null;
 	}
 	
+	@Override
+	protected Reader getContentsAsText(SourceRelativeURI uri, IProject project) {
+		InputStream contents = getContents(uri, project);
+		if (contents != null)
+			return new InputStreamReader(contents, Charsets.UTF_8);
```

Comment:
Same question: should this be the encoding that can be obtained from the project?